pkg/api/v1/resource_helpers.go

```diff
@@ -62,22 +62,13 @@ func (self *ResourceList) NvidiaGPU() *resource.Quantity {
 	return &resource.Quantity{}
 }
 
-func GetContainerStatus(statuses []ContainerStatus, name string) (ContainerStatus, bool) {
+func GetContainerStatusInfo(statuses []ContainerStatus) map[string]ContainerStatus {
```

Comment:
ContainerStatus here should never be more than 2-3 items in general.  The map uses far more memory and creating the map is slower than iterating over a small array, usually.

Do you have a benchmark showing a performance improvement?